core_plugins/reporting/src/index.ts

```diff
@@ -1,29 +1,29 @@
-import { KibanaFunctionalPlugin } from 'kbn-types';
+import { KibanaPlugin } from 'kbn-types';
 import { XPackPluginType } from 'kibana-plugin-xpack';
 import { ReportingConfig } from './ReportingConfig';
 
-export const configPath = ['xpack', 'reporting'];
-
-export const dependencies = ['xpack'];
-
-export const plugin: KibanaFunctionalPlugin<XPackPluginType> = (
-  kibana,
-  dependencies
-) => {
-  const { xpack } = dependencies;
-
-  const log = kibana.logger.get();
-
-  const config$ = kibana.config.createIfExists(ReportingConfig);
-
-  // just an example
-  const isEnabled$ = config$.map(config => config && config.enabled);
-
-  isEnabled$.subscribe(isEnabled => {
-    log.info(`reporting enabled? [${isEnabled}]`);
-  });
-
-  xpack.config$.subscribe(xpackConfig => {
-    log.info(`xpack polling frequency: [${xpackConfig.pollingFrequency}]`);
-  });
-};
+export const plugin: KibanaPlugin<XPackPluginType> = {
+  configPath: ['xpack', 'reporting'],
+  dependencies: ['xpack'],
+  plugin: (
+    kibana,
+    dependencies
+  ) => {
+    const { xpack } = dependencies;
+  
+    const log = kibana.logger.get();
+  
+    const config$ = kibana.config.createIfExists(ReportingConfig);
+  
+    // just an example
+    const isEnabled$ = config$.map(config => config && config.enabled);
+  
+    isEnabled$.subscribe(isEnabled => {
+      log.info(`reporting enabled? [${isEnabled}]`);
+    });
+  
+    xpack.config$.subscribe(xpackConfig => {
+      log.info(`xpack polling frequency: [${xpackConfig.pollingFrequency}]`);
+    });
+  }
+}
```

Comment:
__nit:__ `;` (here and in other plugins)

Reply:
To not clutter this PR I'll fix this in a follow-up. Prettier doesn't actually run on `core_plugins/` right now it looks like, so there are quite a few other minor fixes too.

Reply:
Ok